controlplane/pkg/common/excluded_prefixes_service.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2019 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connectioncontext"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/sdk/prefix_pool"
+)
+
+// ConnectionService makes basic Mechanism selection for the incoming connection
+type excludedPrefixesService struct {
+	prefixes prefix_pool.PrefixPool
+}
+
+// NewExcludedPrefixesService -  creates a service to select endpoint.
+func NewExcludedPrefixesService() networkservice.NetworkServiceServer {
+	return NewExcludedPrefixesServiceFromPath(prefix_pool.PrefixesFilePathDefault)
+}
+
+// NewExcludedPrefixesService -  creates a service to select endpoint.
+func NewExcludedPrefixesServiceFromPath(configPath string) networkservice.NetworkServiceServer {
+	return &excludedPrefixesService{
+		prefixes: prefix_pool.NewPrefixPoolReader(configPath),
+	}
+}
+
+func (eps *excludedPrefixesService) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+	logger := Log(ctx)
+
+	if request.GetConnection() == nil {
+		return nil, errors.Errorf("request's connection cannot be empty")
+	}
+
+	// 7.1.6 Update Request with exclude_prefixes, etc
```

Comment:
Comment 7.1.6 do not make any sense anymore :-)

Reply:
- [ ] remove comment

Reply:
- [x] remove comment